src/cmd/services/m3coordinator/handler/placement/delete_all.go

```diff
@@ -39,16 +40,29 @@ const (
 type deleteAllHandler Handler
 
 // NewDeleteAllHandler returns a new instance of a placement delete all handler.
-func NewDeleteAllHandler(service placement.Service) http.Handler {
-	return &deleteAllHandler{service: service}
+func NewDeleteAllHandler(client clusterclient.Client, cfg config.Configuration) http.Handler {
+	return &deleteAllHandler{client: client, cfg: cfg}
 }
 
 func (h *deleteAllHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	logger := logging.WithContext(ctx)
 
-	if err := h.service.Delete(); err != nil {
+	service, err := Service(h.client, h.cfg)
+	if err != nil {
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	if err := service.Delete(); err != nil {
 		logger.Error("unable to delete placement", zap.Any("error", err))
 		handler.Error(w, err, http.StatusInternalServerError)
+		return
 	}
+
+	json.NewEncoder(w).Encode(struct {
```

Comment:
I get the rationale for this, but it's a little asymmetrical because we don't return something like `Deleted: false` for when we fail to delete. I was thinking to just rely on the implicit `200 OK` to notify the client that it went through.